cimgui_funcs.go

```diff
@@ -1087,21 +1141,21 @@ func (self ImGuiTextFilter) Destroy() {
 }
 
 func (self ImGuiViewport) GetCenter() ImVec2 {
-	pOut := ImVec2{}
-	pOutArg, pOutFin := pOut.wrap()
+	pOut := &ImVec2{}
+	pOutArg, pOutFin := wrap[C.ImVec2, *ImVec2](pOut)
 	defer pOutFin()
 
 	C.Viewport_GetCenter(pOutArg, self.handle())
-	return pOut
+	return *pOut
 }
 
 func (self ImGuiViewport) GetWorkCenter() ImVec2 {
-	pOut := ImVec2{}
-	pOutArg, pOutFin := pOut.wrap()
+	pOut := &ImVec2{}
+	pOutArg, pOutFin := wrap[C.ImVec2, *ImVec2](pOut)
 	defer pOutFin()
 
 	C.Viewport_GetWorkCenter(pOutArg, self.handle())
-	return pOut
+	return *pOut
```

Comment:
Thinking about this more, can't this be like this?

```go
    pOut := ImVec2{}
	pOutArg, pOutFin := wrap[C.ImVec2, *ImVec2](&pOut)
	defer pOutFin()

	C.Viewport_GetWorkCenter(pOutArg, self.handle())
	return pOut
```

This might lead to more efficient code generation

Reply:
> Thinking about this more, can't this be like this?
> 
> ```go
>     pOut := ImVec2{}
>     pOutArg, pOutFin := wrap[C.ImVec2, *ImVec2](&pOut)
>     defer pOutFin()
> 
>     C.Viewport_GetWorkCenter(pOutArg, self.handle())
>     return pOut
> ```
> 
> This might lead to more efficient code generation

sadly, it is not possible in the current implementation...

```go
                        // temporary out arg definition
                        sb.WriteString(fmt.Sprintf("%s := &%s{}\n", outArg.Name, outArgT))

                        argInvokeStmt := argStmtFunc()

```

we do a kind of trick there: to match expected name (and value type) of this out argument.

The current implementation does not allow me to interfer with this name and type at all, so I need to match EXACTLY what it is expected to be.

I think, that this PR is large enough anyway, so (in my opinion) we shouldn't do this here, since (in fact) it is entire new bug/enhancement to deal with.

Reply:
I see, fair enough.